salt/loader.py

```diff
@@ -1224,19 +1228,32 @@ def refresh_file_mapping(self):
         self.file_mapping = salt.utils.odict.OrderedDict()
 
         for mod_dir in self.module_dirs:
-            files = []
             try:
-                # Make sure we have a sorted listdir in order to have expectable override results
+                # Make sure we have a sorted listdir in order to have
+                # expectable override results
                 files = sorted(os.listdir(mod_dir))
             except OSError:
                 continue  # Next mod_dir
+            if six.PY3:
+                try:
+                    pycache_files = [
+                        os.path.join('__pycache__', x) for x in
+                        sorted(os.listdir(os.path.join(mod_dir, '__pycache__')))
```

Comment:
I'm not sure why these are being sorted?

Reply:
I copypasta'ed from [here](https://github.com/saltstack/salt/pull/45095/files#diff-f6777cb7fca1276c97cd4b4a6b9f085aR1234). Perhaps sorting can be removed in both places?

Reply:
If I recall, we are sorting because that way the loader operates in a deterministic way with each loading operation.

Reply:
Ahh yes, I do remember discussing this, many moons ago 😄